crates/collator-selection/src/lib.rs

```diff
@@ -428,25 +428,31 @@ pub mod pallet {
         }
 
         /// Kicks out candidates that did not produce a block in the kick threshold
-        /// and refund their deposits.
+        /// or whose free balance drops below the minimum and refund their deposits.
         pub fn kick_stale_candidates(candidates: Vec<CandidateInfo<T::AccountId, BalanceOf<T>>>) -> Vec<T::AccountId> {
-            // TODO: also kick candidates when escrow balance drops
             let now = frame_system::Pallet::<T>::block_number();
             let kick_threshold = T::KickThreshold::get();
+            let candidacy_bond = Self::candidacy_bond();
             candidates
                 .into_iter()
                 .filter_map(|c| {
                     let last_block = <LastAuthoredBlock<T>>::get(c.who.clone());
                     let since_last = now.saturating_sub(last_block);
-                    if since_last < kick_threshold || Self::candidates().len() as u32 <= T::MinCandidates::get() {
+                    // total balance excludes reserved
+                    let balance = T::StakingCurrency::total_balance(&c.who);
+                    if Self::candidates().len() as u32 <= T::MinCandidates::get() {
+                        // don't kick if not enough candidates
                         Some(c.who)
-                    } else {
+                    } else if since_last >= kick_threshold || balance < candidacy_bond {
+                        // kick if stale OR balance is not sufficient
                         let outcome = Self::try_remove_candidate(&c.who);
                         if let Err(why) = outcome {
                             log::warn!("Failed to remove candidate {:?}", why);
                             debug_assert!(false, "failed to remove candidate {:?}", why);
                         }
                         None
+                    } else {
+                        Some(c.who)
```

Comment:
nit: can be merged with the first branch. The condition would be a bit complex, but you could refactor to something like 
```rust
let surplus_candidates = Self::candidates().len() as u32 > T::MinCandidates::get();
let is_good_collator = since_last < kick_threshold && balance >= candidacy_bond;
if is_good_collator || !surplus_candidates  {
    Some(c.who)
} else {
    None
}
```